// steam-session/dist/helpers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultUserAgent = exports.getSpoofedHostname = exports.isJwtValidForAudience = exports.decodeJwt = exports.API_HEADERS = exports.eresultError = void 0;
const user_agents_1 = require("@doctormckay/user-agents");
const crypto_1 = require("crypto");
const os_1 = require("os");
const EResult_1 = __importDefault(require("./enums-steam/EResult"));
function eresultError(result, errorMessage) {
    let resultMsg = result.toString(); // this is the numeric value, as a string
    resultMsg = EResult_1.default[resultMsg] || resultMsg; // this is now the string representation of the EResult value
    let err = new Error(errorMessage || resultMsg);
    // @ts-ignore
    err.eresult = result;
    return err;
}
exports.eresultError = eresultError;
exports.API_HEADERS = {
    accept: 'application/json, text/plain, */*',
    'sec-fetch-site': 'cross-site',
    'sec-fetch-mode': 'cors',
    'sec-fetch-dest': 'empty'
};
function decodeJwt(jwt) {
    let parts = jwt.split('.');
    if (parts.length != 3) {
        throw new Error('Invalid JWT');
    }
    let standardBase64 = parts[1].replace(/-/g, '+')
        .replace(/_/g, '/');
    return JSON.parse(Buffer.from(standardBase64, 'base64').toString('utf8'));
}
exports.decodeJwt = decodeJwt;
function isJwtValidForAudience(jwt, audience, steamId) {
    let decodedToken;
    try {
        decodedToken = decodeJwt(jwt);
    }
    catch (ex) {
        return false;
    }
    // Check if the steamid matches
    if (steamId && decodedToken.sub != steamId) {
        return false;
    }
    return (decodedToken.aud || []).includes(audience);
}
exports.isJwtValidForAudience = isJwtValidForAudience;
function getSpoofedHostname() {
    let hash = (0, crypto_1.createHash)('sha1');
    hash.update((0, os_1.hostname)());
    let sha1 = hash.digest();
    const CHARS = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
    let output = 'DESKTOP-';
    for (let i = 0; i < 7; i++) {
        output += CHARS[sha1[i] % CHARS.length];
    }
    return output;
}
exports.getSpoofedHostname = getSpoofedHostname;
function defaultUserAgent() {
    return (0, user_agents_1.chrome)();
}
exports.defaultUserAgent = defaultUserAgent;
